### PR TITLE
add hide handlers to the controllers

### PR DESCRIPTION
**Pull request by enigosi**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,13 +7,17 @@ import Contoller, { PRIORITIES } from './src/controller';
...
+};
 
 class ModalController extends React.Component {
   render() {
     return <Contoller {...this.props} ref={(ref) => { modalRef = ref; }} />;
   }
 }
 
+export const hideTopModal = (...args) => modalRef.hideTopModal(...args);
+export const hideAllModals = (...args) => modalRef.hideAllModals(...args);
```
Comment (kirkness) [inline index.js]: Not sure about this. What would be the use case for hiding the modals from out side the scope of the modal itself? We currently lack the ability to close the modal from within the modal i.e. we should pass a prop to the modal `closeModal`. Would this solve the problem this PR is trying to solve?

File: src/controller.js [hunk 2]
```diff
@@ -47,6 +48,30 @@ class Controller extends React.Component {
...
+
+  hideModal = (modalIndex) => {
+    const modals = [...this.state.modals];
+    modals[modalIndex] = {
+      ...modals[modalIndex],
+      isVisible: false,
+    }
+    this.setState({
+      modals
+    }, this.removeBackdropIfStackEmpty())
```
Comment (kirkness) [inline src/controller.js]:
```
this.setState({
     modals
}, this.removeBackdropIfStackEmpty)
```
?

File: index.js (hunk 1 above)
Comment (enigosi) [reply index.js]: I'm doing it as well. I think that if there is `showModal`, `hideModal` also should be exposed for consistency. Although, I can't think now about a valid use case

File: src/controller.js (hunk 2 above)
Comment (enigosi) [reply src/controller.js]: It's setting modals state prop to updated modals arrray and later is firing `removeBackdropIfStackEmpty`, which checks if there is any open modal on the stack

Comment (kirkness) [reply src/controller.js]: I was just meaning that in your code its invoked immedietly

Comment (enigosi) [reply src/controller.js]: I missed it. Good catch

**Pull request merged**